xarray/core/indexing.py

```diff
@@ -614,29 +617,30 @@ def __init__(self, array: Any, key: ExplicitIndexer | None = None):
         self.array = as_indexable(array)
         self.key = key
 
+        shape = []
+        for size, k in zip(self.array.shape, self.key.tuple):
+            if isinstance(k, slice):
+                shape.append(len(range(*k.indices(size))))
+            elif isinstance(k, np.ndarray):
+                shape.append(k.size)
+        self._shape = tuple(shape)
```

Comment:
nit: should this just set `self.shape` and we do away with the `property`?

Reply:
yes that is fine.

Reply:
this doesn't work since the NDArray Mixin defines a `shape` property that we have to override.